Automattic/jetpack
# Connection: fix connection screen for non-admins

#### Changes proposed in this Pull Request:
- fix the bug with connection screen not being available for non-admins
- fix the bug with `/connection/authorize_url` endpoint not being available to non-admins
- only show connection screen if the site already has a connection owner

#### Jetpack product discussion
p1634233299011100-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Connect Jetpack, create a non-admin user (e.g. Editor) and log in using that user.
2. Go to Jetpack Dashboard, confirm that you see the "Connect your WP.com account" bar in the top of the page.
3. Click on the button. Confirm that the connection screen appeared, and the "Connect your WP.com account" bar disappeared.
4. Click "Connect your user account" on the connection screen, go through the connection process.
5. Confirm that you get connected successfully. _There's an odd bug where you get automatically redirected to Jetpack Dashboard from the Calypso Plans page, we'll need to look into it separately. If you encounter it, please confirm that you also see it._
6. Disconnect Jetpack, use admin account to connect it in site-only mode.
7. Go back to the non-admin Jetpack Dashboard. Confirm that you don't see the "Connect your WP.com account" button.
8. Manually switch to the `/wp-admin/admin.php?page=jetpack#/connect-user` page and confirm that you get immediately redirected to `?page=jetpack#/dashboard`.
